### ISSUE TITLE
[Bug] Microsoft.Identity.Web should not use HttpContext.User in AddAccountToCacheFromAuthorizationCodeAsync

### ISSUE THREAD

**Issue by jmprieur** [AzureAD/microsoft-identity-web#235]
**Which Version of Microsoft Identity Web are you using ?**.
Microsoft Identity Web 0.1.5-preview 

**Where is the issue?**
* Web App
    * [x ] Sign-in users and call web APIs
* Token cache serialization
     * [x ] In Memory caches
     * [x ] Session caches
     * [x ] Distributed caches

**Repro**
`TokenAcquisition.AddAccountToCacheFromAuthorizationCodeAsync` uses HttpContext.User whereas it’s not populated yet. Therefore we add some claims to the HttpContext.User.Identity coming from the ID Token:
https://github.com/AzureAD/microsoft-identity-web/blob/0b628e2a56e6525bbca682c50a7528da4d09cc22/src/Microsoft.Identity.Web/TokenAcquisition.cs#L132-L135

We do that so that the token cache serialization can compute a cacheKey. See:

https://github.com/AzureAD/microsoft-identity-web/blob/b9457a826dc4f6d7fbafc67d37f06b2d6cd2acd3/src/Microsoft.Identity.Web/TokenCacheProviders/MsalAbstractTokenCacheProvider.cs#L73

**Expected behavior**
The default user is an anonymous identity (the HttpContext.User is replaced when the token is processed). As an Auth library, we should avoid accessing HttpContext.User anwyway. This field is only updated at specific times in the flow of the request, and with the default scheme of the request. It gets confusing when people are making cookies in your app, and that might not be the right identity in the scenario. 

**Prefered Solution**

- Have this cache key computed and surfaced in the `TokenCacheNotificationArgs` (Expose a new string SuggestedCacheKey in `TokenCacheNotificationArgs`). See https://github.com/AzureAD/microsoft-authentication-library-for-dotnet/issues/1902

This means that we would no longer need to compute the cache key.

**Alternative solution**

- otherwise add a new `.WithTokenSerializationCacheKey(msalAccountId)` method to the acquire token buidlers for confidentials client apps.

The code in TokenAcquisition.cs could be use a new modifier `.WithTokenSerializationCacheKey(msalAccountId)`

```CSharp
 public async Task AddAccountToCacheFromAuthorizationCodeAsync(
            AuthorizationCodeReceivedContext context,
            IEnumerable<string> scopes)
{
 string msalAccountId = context.Principal.GetMsalAccountId();

 var result = await _application
                    .AcquireTokenByAuthorizationCode(scopes.Except(_scopesRequestedByMsal), 
 context.ProtocolMessage.Code)
                    .WithSendX5C(_microsoftIdentityOptions.SendX5C)
                    .WithTokenSerializationCacheKey(msalAccountId)
                    .ExecuteAsync()
                    .ConfigureAwait(false);
}
```
Then this code could be simplified to take info from the `TokenCacheNotificationArgs`

https://github.com/AzureAD/microsoft-identity-web/blob/b9457a826dc4f6d7fbafc67d37f06b2d6cd2acd3/src/Microsoft.Identity.Web/TokenCacheProviders/MsalAbstractTokenCacheProvider.cs#L60-L75

**Comment by jennyf19** [comment]
@jmprieur doing this in 3 parts. 
[1] Waiting on a [PR](https://github.com/AzureAD/microsoft-authentication-library-for-dotnet/pull/1908) to merge in MSAL .NET
[2] Continue the cache key work w/new package, already have started w/this [branch](https://github.com/AzureAD/microsoft-identity-web/tree/jennyf/cacheKey)
[3] Address the claims issue

**Comment by jennyf19** [comment]
Included in [0.2.0-preview release](https://github.com/AzureAD/microsoft-identity-web/releases/tag/0.2.0-preview)
